plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaFilterManager.java

```diff
@@ -96,23 +96,16 @@ public KafkaFilteringResult getKafkaFilterResult(
             Set<Long> partitionIdsFiltered = partitionIds;
             Optional<Map<ColumnHandle, Domain>> domains = constraint.getDomains();
 
-            for (Map.Entry<ColumnHandle, Domain> entry : domains.get().entrySet()) {
+            for (Map.Entry<ColumnHandle, Domain> entry : domains.orElseThrow().entrySet()) {
                 KafkaColumnHandle columnHandle = (KafkaColumnHandle) entry.getKey();
                 if (!columnHandle.isInternal()) {
                     continue;
                 }
+                Domain domain = entry.getValue();
                 switch (columnHandle.getName()) {
-                    case PARTITION_OFFSET_FIELD:
-                        offsetRanged = filterRangeByDomain(entry.getValue());
-                        break;
-                    case PARTITION_ID_FIELD:
-                        partitionIdsFiltered = filterValuesByDomain(entry.getValue(), partitionIds);
-                        break;
-                    case OFFSET_TIMESTAMP_FIELD:
-                        offsetTimestampRanged = filterRangeByDomain(entry.getValue());
-                        break;
-                    default:
-                        break;
+                    case PARTITION_OFFSET_FIELD -> offsetRanged = filterRangeByDomain(domain);
+                    case PARTITION_ID_FIELD -> partitionIdsFiltered = filterValuesByDomain(domain, partitionIds);
+                    case OFFSET_TIMESTAMP_FIELD -> offsetTimestampRanged = filterRangeByDomain(domain);
```

Comment:
`(coding) InnerAssignment: Inner assignments should be avoided.`
Don't use switch expression instead of switch statement.